spark3/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -168,9 +177,9 @@ private boolean isWapTable() {
 
   // the writer factory works for both batch and streaming
   private WriterFactory createWriterFactory() {
-    return new WriterFactory(
-        table.spec(), format, table.locationProvider(), table.properties(), io, encryptionManager, targetFileSize,
-        writeSchema, dsSchema, partitionedFanoutEnabled);
+    // broadcast the table metadata as the writer factory will be sent to executors
+    Broadcast<Table> staticTable = lazySparkContext().broadcast(SparkUtil.toStaticTable(table));
```

Comment:
@pvary, is there a broadcast concept in Hive too? We had some performance issues serializing/deserializing FileIO.

Reply:
Not that I know off 😢 
We pass these variables as part of the configuration object for the Mappers/Reducers separately.
We are not yet on this level in the optimizations, but soon... 😄